From 5b26d8a2da4c8bba72781e45ec89572843f55f35 Mon Sep 17 00:00:00 2001
From: Mike Baiocchi <baiocchi@us.ibm.com>
Date: Tue, 21 Apr 2015 13:23:36 -0500
Subject: [PATCH] Fix Bug in mss_setup_bars

There was an update made to mss_setup_bars.C to fix memory corruption, but
this commit corrects an issue that was missed by this initial update.

Change-Id: I32889a5d07b33f5496e5ea12ec50a0f26d0e1561
CQ: SW296125
---
 src/usr/hwpf/hwp/dram_initialization/proc_setup_bars/mss_setup_bars.C | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/usr/hwpf/hwp/dram_initialization/proc_setup_bars/mss_setup_bars.C b/src/usr/hwpf/hwp/dram_initialization/proc_setup_bars/mss_setup_bars.C
index 274dfdc..6c58193 100644
--- a/src/usr/hwpf/hwp/dram_initialization/proc_setup_bars/mss_setup_bars.C
+++ b/src/usr/hwpf/hwp/dram_initialization/proc_setup_bars/mss_setup_bars.C
@@ -615,7 +615,7 @@ fapi::ReturnCode mss_setup_bars(const fapi::Target& i_pu_target,   std::vector<f
                     if(mcs_sz !=  sizeInfo.MCS_size[group_data[i][j]])
                     {
                           FAPI_INF(" Group %zd will not be configured as MCS %d is not valid grouped size is %d , present MCS size is %d \n",i,group_data[i][j],mcs_sz, sizeInfo.MCS_size[group_data[i][j]]);
-                          for(uint8_t k = 0; k<32;k++) { group_data[i][k]=0; }
+                          for(uint8_t k = 0; k<16;k++) { group_data[i][k]=0; }
                      }
                }
            }
-- 
1.8.2.2

